File: vibecheck-backend/VibeCheck.DAL/Entities/Song.cs
namespace VibeCheck.DAL.Entities
{
    public class Song
    {
        public Guid SongId { get; set; }
        public string DeezerId { get; set; } // Deezer song ID
        public Guid RoundId { get; set; } 
        public string SongTitle { get; set; } = string.Empty;
        public string Artist { get; set; } = string.Empty; 
        public DateTime SubmittedAt { get; set; }

        // New properties from frontend
        public string? AlbumName { get; set; }
        public string? AlbumCoverSmall { get; set; }
        public string? AlbumCoverBig { get; set; }
        public string? PreviewUrl { get; set; }
        
        // Navigation properties
        public Round Round { get; set; } = null!;
        public List<User> Users { get; set; } = null!;
        public ICollection<Vote> Votes { get; set; } = null!; // changed in num of votes or delete
    }
}

File: vibecheck-frontend/src/types/song.ts
import { User, UserDto } from "./user";
import { Vote } from "./vote";

export interface Song {
  id: string;
  title: string;
  previewUrl: string;
  artistName: string;
  albumName: string;
  albumCoverSmall: string;
  albumCoverBig: string;
  votes?: Vote[];
  users?: User[];
}

export interface SongDto {
  id: string;
  deezerSongId: string;
  title: string;
  previewUrl: string;
  artistName: string;
  albumName: string;
  albumCoverSmall: string;
  albumCoverBig: string;
  voteCount: number;
  votes?: Vote[];
  users?: UserDto[];
}

export interface SongSubmission {
  id: string;
  deezerSongId: string;
  song: SongDto;
  users: UserDto[];
  votes: number;
  hasUserVoted: boolean;
  votedBy?: string[];
  isUserSubmitter?: boolean;
}